cli/src/main/java/com/devonfw/tools/ide/tool/plugin/PluginBasedCommandlet.java

```diff
@@ -91,37 +92,69 @@ public Path getPluginsInstallationPath() {
     return this.context.getPluginsPath().resolve(this.tool);
   }
 
+  @Override
+  protected void postInstall(boolean newlyInstalled) {
+
+    super.postInstall(newlyInstalled);
+    if (newlyInstalled) {
+      Path pluginsInstallationPath = getPluginsInstallationPath();
+      FileAccess fileAccess = this.context.getFileAccess();
+      fileAccess.delete(pluginsInstallationPath);
+      fileAccess.mkdirs(pluginsInstallationPath);
+      installPlugins(getPlugins().getPlugins());
+    }
+    // else ? check all plugins are installed, retry failed plugin installations ?
+  }
+
   /**
-   * @param plugin the {@link PluginDescriptor} to install.
+   * Method to install active plugins or to handle install for inactive plugins
+   *
+   * @param plugins as {@link Collection} of plugins to install.
    */
-  public abstract void installPlugin(PluginDescriptor plugin);
+  protected void installPlugins(Collection<ToolPluginDescriptor> plugins) {
+    for (ToolPluginDescriptor plugin : plugins) {
+      if (plugin.active()) {
+        try (Step step = this.context.newStep("Install plugin " + plugin.name())) {
+          installPlugin(plugin, step);
+        }
+      } else {
+        handleInstall4InactivePlugin(plugin);
```

Comment:
I did not introduce this method in this PR. What is wrong with the naming?
Do you want to write `For` instead of `4`?
I am also happy with removing the `4` but not tonight and not going to block this PR for it...